core/src/test/java/org/apache/iceberg/actions/TestBinPackRewriteFilePlanner.java

```diff
@@ -290,7 +290,8 @@ void testValidOptions() {
                 BinPackRewriteFilePlanner.MAX_FILE_GROUP_SIZE_BYTES,
                 BinPackRewriteFilePlanner.DELETE_FILE_THRESHOLD,
                 BinPackRewriteFilePlanner.DELETE_RATIO_THRESHOLD,
-                RewriteDataFiles.REWRITE_JOB_ORDER));
+                RewriteDataFiles.REWRITE_JOB_ORDER,
+                RewriteDataFiles.MAX_FILES_TO_REWRITE));
```

Comment:
We need tests in the core to test that the filtering working as expected

Reply:
Thank you for the review . I moved tests from spark module to core `TestBinPackRewriteFilePlanner` class